### PR TITLE
Migration from jar to aar

### PR DESCRIPTION
**Pull request by dario-budimir-aevi**
* Migrated to new AGP
* Migrated to AndroidX
* misc cleanup and fixes

### CODE REVIEW

File: .github/workflows/deploy-tagged-release.yml [hunk 1]
```diff
@@ -112,9 +115,9 @@ jobs:
           -Pversion_suffix=${{env.VERSION_SUFFIX}} \
           -Pversion_build_counter=${{env.VERSION_BUILD_NUMBER}} \
 
-      - name: Build and upload library jars to maven repo
+      - name: Build and upload library jars to Github repo
```
Comment (andreas-borglin) [inline .github/workflows/deploy-tagged-release.yml:118]: aars

File: .github/workflows/pr-build-test.yml [hunk 2]
```diff
@@ -5,6 +5,8 @@ env:
   AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
   GITHUB_ACTOR: ${{ secrets.CI_BOT_USERNAME }}
   GITHUB_TOKEN: ${{ secrets.CI_BOT_PAT }}
+  SDK_DEV_KEYSTORE: ${{ secrets.SDK_DEV_KEYSTORE }}
```
Comment (andreas-borglin) [inline .github/workflows/pr-build-test.yml:8]: Don't need these here

File: .github/workflows/deploy-tagged-release.yml [hunk 3]
```diff
@@ -5,6 +5,8 @@ env:
   AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
   GITHUB_ACTOR: ${{ secrets.CI_BOT_USERNAME }}
   GITHUB_TOKEN: ${{ secrets.CI_BOT_PAT }}
+  SDK_DEV_KEYSTORE: ${{ secrets.SDK_DEV_KEYSTORE }}
```
Comment (andreas-borglin) [inline .github/workflows/deploy-tagged-release.yml:8]: Don't need these here

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.3.0] - 2022-08-25
+
+* Migration from jar to aar
+* Migrated to new AGP
+* Migrated to AndroidX
+* [PAYINIT-50](https://jira.aevi.com/browse/PAYINIT-50): Updating test framework in AppFlow stack
```
Comment (andreas-borglin) [inline CHANGELOG.md:12]: Don't reference JIRA here - it's a public changelog

File: FlowServiceSample/build.gradle [hunk 5]
```diff
@@ -2,13 +2,17 @@ apply plugin: 'com.android.application'
...
-ext.applicationTargetSdkVersion = 28
-ext.applicationMinSdkVersion = 21
 
 apply from: gradleScript('android/artifacts.gradle')
 apply from: gradleScript('android/versioning.gradle')
 apply from: gradleScript('android/basic-android.gradle')
 
+configurations {
+    all {
+        exclude group: 'androidx.lifecycle', module: 'lifecycle-viewmodel-ktx'
```
Comment (andreas-borglin) [inline FlowServiceSample/build.gradle:12]: Why?

File: build.gradle [hunk 6]
```diff
@@ -1,125 +1,40 @@
...
-
 allprojects {
     repositories {
+        google()
+        mavenCentral()
         maven { url 'https://jitpack.io' }
     }
 }
 
-task cleanDocs(type: Delete) {
```
Comment (andreas-borglin) [inline build.gradle:38]: This can't be deleted - it's required for how we generate javadocs

File: uninstallSamples.sh [hunk 7]
```diff
@@ -1,3 +0,0 @@
-adb uninstall com.aevi.sdk.pos.flow.paymentinitiationsample
```
Comment (andreas-borglin) [inline uninstallSamples.sh:1]: This was added for convenience - not sure it should be deleted

File: dist.gradle [hunk 8]
```diff
@@ -1,7 +0,0 @@
-task publishJavadocsAsLatest(type: Exec, dependsOn: javadocs) {
```
Comment (andreas-borglin) [inline dist.gradle:1]: This should absolutely not be deleted..it's how we publish the javadocs!

File: installSamples.sh [hunk 9]
```diff
@@ -1,4 +0,0 @@
-./gradlew :PaymentServiceSample:installRelease
```
Comment (andreas-borglin) [inline installSamples.sh:1]: This is still useful to have

File: FlowServiceSample/build.gradle (hunk 5 above)
Comment (dario-budimir-aevi) [reply FlowServiceSample/build.gradle:12]: this is regarding duplicate classes found in different libraries so we need to exclude them and use only one

Comment (dario-budimir-aevi) [reply FlowServiceSample/build.gradle:12]:
```
Duplicate class androidx.lifecycle.ViewTreeViewModelKt found in modules jetified-lifecycle-viewmodel-ktx-2.3.1-runtime (androidx.lifecycle:lifecycle-viewmodel-ktx:2.3.1) and lifecycle-viewmodel-2.5.0-runtime (androidx.lifecycle:lifecycle-viewmodel:2.5.0)
```

**Pull request closed without merging**